moldesign/models/lammps.py

```diff
@@ -0,0 +1,370 @@
+    
+    PARAMETERS = [
+        mdt.parameters.Parameter(
+            'skin', 'Extra distance beyond force cutoff',
+            type=u.angstrom, default=2.0)]
+
+    
+    # calculate potential energy and force
+    def calculate(self, requests):
+        
+        lammps_system = self.lammps_system
+        lammps_system.run(500)
+        pe = lammps_system.eval("pe")
+        forceArray = []
+        for i in range(0, lammps_system.atoms.natoms):
+            forceArray.append(self.lammps_system.atoms[i].force)
+        return {'potential_energy': pe,
+                'forces': forceArray}
+        
+    def prep(self, force=False):
+        """
+        Drive the construction of the LAMMPS simulation
+        This will rebuild this OpenMM simulation if: A) it's not built yet, or B)
+        there's a new integrator
+        """
+        self._create_system()
+        self._prepped = True
+        
+        # TODO: wiring integrator
+
+    
+    #################################################
+    # "Private" methods for managing LAMMPS are below
+    
+    def _create_system(self):
+        
+        # Ensure force fields are assigned 
+        if 'amber_params' not in self.mol.ff:
+            raise NotImplementedError('Assign forcefield parameters to the system')
+
+        # Store the PRMTOP file if ff are assigned and use it to create lammps molecule data
+        self.mol.ff.amber_params.prmtop.put('/tmp/tmp.prmtop')
+
+        import parmed as med
+        parmedmol = med.load_file('/tmp/tmp.prmtop')
+
+        lmp = lammps()
+        L = PyLammps(ptr=lmp)
+        
+        dataPath = _create_lammps_data(parmedmol)
+        print 'Created LAMMPS data'
+
+        L.command("units real")
+        L.command("atom_style full")
+        L.command("pair_style lj/charmm/coul/long 8.0 10.0 10.0")
+        L.command("bond_style harmonic")
+        L.command("angle_style harmonic")
+        L.command("dihedral_style charmm")
+        L.command("improper_style harmonic")
+        L.command("kspace_style pppm 0.0001")
+        L.command("read_data " + dataPath)
+        
+        # NOTE: Do we want this?
+        L.command("neighbor " + str(self.params.skin) + " bin")
+        
+        L.command("thermo_style custom step temp pe etotal")
+        # NOTE: By default, time step is 10.0
+        L.command("thermo 10")
+        
+        # group hbonds
+        hbond_lj_types = _group_hbonds(parmedmol)
+        if len(hbond_lj_types) > 0:
+            L.command("group hbond type " + hbond_lj_types)
+            self.group_hbond = True
+        else:
+            self.group_hbond = False
+
+        # group water
+        water_res = _group_water(parmedmol)
+        if len(water_res) > 0:
+            L.command("group water molecule " + water_res)
+            self.group_water = True
+        else:
+            self.group_water = False
+
+        self.lammps_system = L
+        print 'Created LAMMPS object'
```

Comment:
We generally want to keep console logging like this pretty light, except in the following situations:
1) We made potentially unexpected changes to the `Molecule` object.
2) The function has a somewhat ambiguous effect that needs to be logged
3) We're in the middle of a long simulation and need to provide status updates

For OpenMMPotential, we log the system creation because it falls into case 2). OpenMM systems bind to a specific hardware interface - the CPU, an nVidia GPU, or a generic GPU - so we need to tell the user which resource the simulation will be using. But here, there's nothing ambiguous, so we don't need to log it.